### PR TITLE
Implement `public Biome getRiver()`

### PR DESCRIPTION
**Pull request by valoeghese**

### CODE REVIEW

File: .recommenders/caches/identified-project-coordinates.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[[{"location":"C:\\Users\\mekal\\.gradle\\caches\\modules-2\\files-2.1\\com.google.guava\\guava\\21.0\\3a3d111be1be1b745edfa7d91678a12d7ed38709\\guava-21.0.jar","type":"JAR","hints":{}},"com.google.guava:guava:21.0.0"],[{"location":"C:\\Program Files\\Java\\jre1.8.0_201","type":"JRE","hints":{"EXECUTION_ENVIRONMENT":"JavaSE-1.8"}},"jre:jre:1.8.0"],[{"location":"C:\\Users\\mekal\\Desktop\\Modding\\Patchwork\\patchwork-api\\patchwork-biomes","type":"PROJECT","hints":{"PROJECT_NAME":"patchwork-biomes"}},"ABSENT"]]
```
Comment (shedaniel) [inline .recommenders/caches/identified-project-coordinates.json]: I don't think the files in `.recommenders` should be added.

File: patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/ForgeBiomeExt.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.impl.biomes;
+
+import net.minecraft.world.biome.Biome;
+
+public interface ForgeBiomeExt {
```
Comment (coderbot16) [inline patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/ForgeBiomeExt.java]: Maybe just `RiverSupplier`?

File: patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/PatchworkBiomes.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	private static void addRivers(Biome biome) {
+		Biome river = ((ForgeBiomeExt) biome).getRiver();
+
+		if (river == null) {
+			failedBiomes.add(biome);
+			return;
+		}
+
+		if (river != getDefaultRiver(biome)) {
```
Comment (coderbot16) [inline patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/PatchworkBiomes.java]: Could you deduplicate this? Maybe add another util method

File: patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/PatchworkBiomes.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		if (biome == Biomes.SNOWY_TUNDRA) {
+			return Biomes.FROZEN_RIVER;
+		} else if (biome == Biomes.MUSHROOM_FIELDS || biome == Biomes.MUSHROOM_FIELD_SHORE) {
+			return Biomes.MUSHROOM_FIELD_SHORE;
+		}
+
+		return Biomes.RIVER;
+	}
+
+	private static Set<Biome> failedBiomes = Sets.newHashSet();
```
Comment (coderbot16) [inline patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/PatchworkBiomes.java]: would prefer this to be at the top of the class

Comment (valoeghese) [reply patchwork-biomes/src/main/java/com/patchworkmc/impl/biomes/PatchworkBiomes.java]: I usually put static fields at the bottom of the class, but this isn't my project so ok

**Comment by valoeghese** [comment]
done

**Pull request merged**